### DISCUSSION TITLE
Consider selling the `required` modifier in `CS8618` warning

### DISCUSSION THREAD

**Discussion by JJLovesLife** [dotnet/roslyn#75947] [Ideas] [answered]
This [required modifier](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/required) is quite a good feature, especially when fixing [CS8618](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/nullable-warnings#nonnullable-reference-not-initialized) warnings.
However, I never know this modifier until this year's .NET Conf [C#'s Best features you might not be using](https://youtu.be/yuXw7oj0Bg0?si=gsFLcsuNRgKQIB6f&t=731). I think it would be great if you can telling people there is a required modifier might fit your purpose in the message of CS8618. A lot of people would like to use this modifier!

**Comment by RikkiGibson** [comment, marked as answer]
The diagnostic message I am seeing is

> warning CS8618: Non-nullable property 'Prop' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring the property as nullable.

[SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQGYACLEgYRIG8CT6zT1sAGEgBQCcB7ABxpIBzAKYAXANwkAzmMkBfOg2JkALCQCyACgCUNRfQX45QA==)

**Reply by JJLovesLife** [reply]
Wow, looks someone else already mentioned that and the message got updated. That's great! https://github.com/dotnet/roslyn/issues/68716
